packages/ai-core/src/browser/prompttemplate-contribution.ts

```diff
@@ -122,6 +135,19 @@ export class PromptTemplateContribution implements LanguageGrammarDefinitionCont
         );
     }
 
+    provideVariableCompletions(model: monaco.editor.ITextModel, position: monaco.Position): ProviderResult<monaco.languages.CompletionList> {
+        return this.getSuggestions(
+            model,
+            position,
+            '#{',
+            this.variableService.getVariables(),
+            monaco.languages.CompletionItemKind.Variable,
+            variable => variable.id,
```

Comment:
```suggestion
            variable => variable.args?.some(arg => !arg.isOptional) ? variable.name + PromptText.VARIABLE_SEPARATOR_CHAR : variable.name,

```

In [`chat-view-language-contribution.ts`](https://github.com/eclipsesource/theia/blob/f1562a2187c0cb644d851352ddcc119aed31c86f/packages/ai-chat-ui/src/browser/chat-view-language-contribution.ts#L137-L138) we actually complete to the `variable.name` and not the id. I think we should make this consistent and use `name` here too.

Even if we don't provide a completion for arguments here (which might be a separate thing to do at some point), it might be worth to still complete the `:` if it has mandatory arguments, to at least indicate that this variable requires an argument.